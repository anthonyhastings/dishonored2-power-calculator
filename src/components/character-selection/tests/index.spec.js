import React from 'react';
import Immutable from 'immutable';
import { render } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import CharacterSelection from '../';

const characters = Immutable.fromJS({
  abc: {
    id: 'aj',
    name: 'Adam Jensen',
    description: 'Task Force 29 operative.'
  },
  def: { id: 'sr', name: 'Scott Ryder', description: 'The pathfinder.' }
});

describe('CharacterSelection component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = render(
      <MemoryRouter>
        <CharacterSelection characters={characters}></CharacterSelection>
      </MemoryRouter>
    );
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
