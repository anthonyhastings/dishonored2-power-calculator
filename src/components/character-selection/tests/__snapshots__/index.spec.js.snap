// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterSelection component when rendered renders a heading with sub-areas for each character 1`] = `
<section
  className="characterSelection__container"
>
  <h1>
    Character Selection
  </h1>
  <div
    className="character__container"
  >
    <h2
      className="character__name"
    >
      Adam Jensen
    </h2>
    <p
      className="character__description"
    >
      Task Force 29 operative.
    </p>
    <MockLink
      to="/dummy-id/powers"
    >
      Choose 
      Adam Jensen
    </MockLink>
  </div>
  <div
    className="character__container"
  >
    <h2
      className="character__name"
    >
      Scott Ryder
    </h2>
    <p
      className="character__description"
    >
      The pathfinder.
    </p>
    <MockLink
      to="/fake-id/powers"
    >
      Choose 
      Scott Ryder
    </MockLink>
  </div>
</section>
`;
